packages/@storefront/breadcrumbs/test/unit/refinement-crumb.ts

```diff
@@ -0,0 +1,95 @@
+import RefinementCrumb from '../../src/refinement-crumb';
+import suite from './_suite';
+
+suite('RefinementCrumb', ({ expect, spy, itShouldProvideAlias }) => {
+  let refinementCrumb: RefinementCrumb;
+
+  beforeEach(() => {
+    refinementCrumb = new RefinementCrumb();
+  });
+
+  itShouldProvideAlias(RefinementCrumb, 'refinementCrumb');
+
+  describe('init()', () => {
+    it('should call updateState', () => {
+      const updateState = refinementCrumb.updateState = spy();
+
+      refinementCrumb.init();
+
+      expect(updateState).to.be.called;
+    });
+  });
+
+  describe('onUpdate()', () => {
+    it('should call updateState', () => {
+      const updateState = refinementCrumb.updateState = spy();
+
+      refinementCrumb.onUpdate();
+
+      expect(updateState).to.be.called;
+    });
+  });
+
+  describe('updateState()', () => {
+    it('should set state', () => {
+      const label = 'test';
+      const props = refinementCrumb.props = <any>{ a: 'b', c: 'd', range: false, value: label };
+
+      refinementCrumb.updateState();
+
+      expect(refinementCrumb.state).to.eql({ ...props, label });
+    });
+
+    it('should set label to a range', () => {
```

Comment:
What happens if range-type refinement doesn't include both a `low` and `high` value?

Reply:
We don't do that assertion anywhere else, so I would say it's out of scope. If we were to do it, we'd probably want to do it for everything else as well.

Reply:
If a range refinement comes back from the DB, it will always have a `low` and `high`. So a range refinement in the store will always have a `low` and `high`. The only way you could end up in a scenario where it doesn't have those properties is if either someone misuses this component, or someone adds a range refinement to the store manually and doesn't provide `low` & `high`. I don't think we should support these cases.

Reply:
But there's a ticket somewhere about validating props which I think accomplishes what you want in this scenario (+ I'm in the process of moving our normalization to the reducers which means that everything that is created in the store will always have the correct shape and types)

Reply:
> But there's a ticket somewhere about validating props which I think accomplishes what you want in this scenario (+ I'm in the process of moving our normalization to the reducers which means that everything that is created in the store will always have the correct shape and types)

Sounds good; marking this as resolved.